sway-core/src/semantic_analysis/ast_node/declaration/auto_impl.rs

```diff
@@ -520,9 +521,22 @@ where
         let mut reads = false;
         let mut writes = false;
 
+        // used to check for name collisions
+        let mut contract_methods: BTreeMap<String, Vec<Span>> = <_>::default();
+
+        // generate code
         for r in contract_fns {
             let decl = engines.de().get(r);
 
+            let name = decl.name.as_str();
+            if !contract_methods.contains_key(name) {
+                contract_methods.insert(name.to_string(), vec![]);
+            }
+            contract_methods
+                .get_mut(name)
+                .unwrap()
+                .push(decl.name.span());
+
```

Comment:
nit: removal of unwrap
```suggestion
        let name = decl.name.as_str();
        contract_methods
            .entry(name.to_string())
            .or_default()
            .push(function_declaration.name.span());
```

Reply:
This allocates a `String` when it is not needed. That is why I went with `contains_key`.